sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClient.java

```diff
@@ -0,0 +1,354 @@
+
+package com.azure.search;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.PagedIterableBase;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.search.models.AutocompleteItem;
+import com.azure.search.models.AutocompleteOptions;
+import com.azure.search.models.IndexBatch;
+import com.azure.search.models.IndexDocumentsResult;
+import com.azure.search.models.RequestOptions;
+import com.azure.search.models.SearchOptions;
+import com.azure.search.models.SearchResult;
+import com.azure.search.models.SuggestOptions;
+import com.azure.search.models.SuggestResult;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+/**
+ * Cognitive Search Synchronous Client to query an index and upload, merge, or delete documents
+ */
+@ServiceClient(builder = SearchIndexClientBuilder.class)
+public class SearchIndexClient {
+
+    private final SearchIndexAsyncClient asyncClient;
+
+    /**
+     * Package private constructor to be used by {@link SearchIndexClientBuilder}
+     *
+     * @param searchIndexAsyncClient Async SearchIndex Client
+     */
+    SearchIndexClient(SearchIndexAsyncClient searchIndexAsyncClient) {
+        this.asyncClient = searchIndexAsyncClient;
+    }
+
+    /**
+     * Gets the name of the Azure Cognitive Search index.
+     *
+     * @return the indexName value.
+     */
+    public String getIndexName() {
+        return asyncClient.getIndexName();
+    }
+
+    /**
+     * Gets the {@link HttpPipeline} powering this client.
+     *
+     * @return the pipeline.
+     */
+    HttpPipeline getHttpPipeline() {
+        return this.asyncClient.getHttpPipeline();
+    }
+
+    /**
+     * Uploads a collection of documents to the target index
+     *
+     * @param documents collection of documents to upload to the target Index.
+     * @return document index result.
+     */
+    public IndexDocumentsResult uploadDocuments(Iterable<?> documents) {
+        return this.uploadDocumentsWithResponse(documents, Context.NONE).getValue();
+    }
+
+    /**
+     * Uploads a collection of documents to the target index
+     *
+     * @param documents collection of documents to upload to the target Index.
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return response containing the document index result.
+     */
+    public Response<IndexDocumentsResult> uploadDocumentsWithResponse(Iterable<?> documents, Context context) {
+        return asyncClient.uploadDocumentsWithResponse(documents, context).block();
+    }
+
+    /**
+     * Merges a collection of documents with existing documents in the target index.
+     *
+     * @param documents collection of documents to be merged
+     * @return document index result
+     */
+    public IndexDocumentsResult mergeDocuments(Iterable<?> documents) {
```

Comment:
@navalev There is important documentation missing from the `merge` and `mergeOrUpload` methods. See the corresponding docs in the .NET SDK:

https://github.com/Azure/azure-sdk-for-net/blob/8f8b4f5452c0521bfb71636730c947000e44b5f8/sdk/search/Microsoft.Azure.Search.Data/src/Customizations/Documents/Models/IndexBatch.Customization.cs#L54

Reply:
documentation added to both sync and async clients